src/ORM/FieldType/DBGeneratedColumn.php

```diff
@@ -0,0 +1,288 @@
+<?php
+
+namespace SilverStripe\ORM\FieldType;
+
+use InvalidArgumentException;
+use SilverStripe\Assets\Storage\DBFile;
+use SilverStripe\Core\Validation\ValidationResult;
+use SilverStripe\Forms\FormField;
+use SilverStripe\Model\ModelData;
+use SilverStripe\ORM\DB;
+use SilverStripe\ORM\Filters\SearchFilter;
+
+/**
+ * A column which gets generated based on some SQL statement.
+ * This can be stored (generated whenever columns in the generation statement are updated), or virtual (generated on the fly when accessed).
+ */
+class DBGeneratedColumn extends DBField
+{
+    public const string GENERATION_STORED = 'STORED';
+
+    public const string GENERATION_VIRTUAL = 'VIRTUAL';
+
+    /**
+     * DBField classes which must not be used as the underlying data representation for a generated column.
+     */
+    private static array $invalid_child_field_types = [
+        DBClassName::class,
+        DBClassNameVarchar::class,
+        DBComposite::class,
+        DBFile::class,
+        DBForeignKey::class,
+        DBGeneratedColumn::class,
+        DBPolymorphicForeignKey::class,
+        DBPrimaryKey::class,
+    ];
+
+    private DBField $childField;
+
+    private string $generationExpression;
+
+    private string $generationType;
+
+    public function __construct(?string $name = null, string $fieldSpec = '', string $expression = '', string $type = DBGeneratedColumn::GENERATION_VIRTUAL)
```

Comment:
```suggestion
    public function __construct(?string $name = null, string $fieldSpec = '', string $sqlExpression = '', string $storageType = DBGenerated::VIRTUAL)
```
And update references

Reply:
Any discussion about these changes will be in the threads where you suggested the original change to the API.

Though I will note there's no need for constructor args to match property args and in many cases I prefer a simpler name in the constructor.... that's super subjective so I'll make changes where the underlying API changed in order to reduce ping pong.